kusto/query_builder.go

```diff
@@ -506,6 +507,185 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
 	return s
 }
 
+// AddDatabase will add a normalized Database string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddDatabase(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddTable will add a normalized Table string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddTable(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddColumn will add a normalized Column string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddColumn(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddFunction will add a normalized Function string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFunction(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// NormalizeName normalizes a string in order to be used safely in the engine.
+func (s Stmt) NormalizeName(query string, forceNormalization bool) Stmt {
+	if query == "" {
+		return s
+	}
+	if !forceNormalization && IsIdentifier(query) {
+		return s
+	}
+	if strings.HasPrefix(query, "[") {
+		return s
+	}
+	if !(strings.Contains(query, "'")) {
+		query = "['" + query + "']"
+	} else {
+		query = "[\"" + query + "\"]"
+	}
+
+	s.queryStr = s.queryStr + query
+	return s
+}
+
+// IsIdentifier checks whether a given string is an identifier
+func IsIdentifier(query string) bool {
+	if query == "" {
+		return false
+	}
+	if !unicode.IsLetter(rune(query[0])) && query[0] != '_' {
+		return false
+	}
+	for _, c := range query {
+		if !(((unicode.IsLetter(c) || unicode.IsDigit(c)) && unicode.In(c, unicode.ASCII_Hex_Digit)) || c == '_') {
+			return false
+		}
+	}
+	return true
+}
+
+// AddQuotedString escapes a string to be safely added to a stmt
+func (s Stmt) AddQuotedString(value string, hidden bool) Stmt {
```

Comment:
You don't use this anywhere in the code. Shouldn't you quote the name in 'NormalizeName'?

Reply:
@AsafMah , the logic is taken from C#.
I agree NormalizeName() smells a bit fishy and has attack paths, but I would like to cover them with Ziv before making changes both here and in C#